dopamine/discrete_domains/run_experiment.py

```diff
@@ -535,3 +539,33 @@ def _save_tensorboard_summaries(self, iteration, num_episodes,
             tag='Train/AverageReturns', simple_value=average_reward),
     ])
     self._summary_writer.add_summary(summary, iteration)
+
+
+@threading_utils.local_attributes(['_environment'])
+class AsyncRunner(Runner):
+  """Defines a train runner for asynchronous training."""
+
+  def __init__(
+      self, base_dir, create_agent_fn,
+      create_environment_fn=atari_lib.create_atari_environment, **kwargs):
+    """Creates an asynchronous runner.
+
+    Args:
+      base_dir: str, the base directory to host all required sub-directories.
+      create_agent_fn: A function that takes as args a Tensorflow session and an
+        environment, and returns an agent.
+      create_environment_fn: A function which receives a problem name and
+        creates a Gym environment for that problem (e.g. an Atari 2600 game).
+      **kwargs: Additional positional arguments.
+    """
+    threading_utils.initialize_local_attributes(
+        self, _environment=create_environment_fn)
+    super(AsyncRunner, self).__init__(
+        base_dir=base_dir, create_agent_fn=create_agent_fn,
+        create_environment_fn=create_environment_fn, **kwargs)
+
+  def _initialize_session(self):
+    """Creates a tf.Session that supports GPU usage in multiple threads."""
+    config = tf.ConfigProto(allow_soft_placement=True)
+    config.gpu_options.allow_growth = True
```

Comment:
Is there a downside to setting `config.gpu_options.allow_growth = True` in the regular `TrainRunner`, too?